[RHACS] ROX-18663: Documenting delegated image scanning feature (#64172)
openshift/openshift-docs: operating/examine-images-for-vulnerabilities.adoc

[id="configuring-environment-variables-after-installation_{context}"]
==== Configuring environment variables after installation

[discrete]
===== OpenShift Container Platform

.Procedure

* Run the following command to enable delegated scanning by setting the value of the environment variable`ROX_DELEGATED_SCANNING_DISABLED` to `true` for a deployment named `sensor`:
+
[source,terminal,subs="+quotes"]
----
$ oc set env deploy/sensor ROX_DELEGATED_SCANNING_DISABLED="true"
----

[discrete]
===== Kubernetes

.Procedure

* Run the following command to enable delegated scanning by setting the value of the environment variable `ROX_DELEGATED_SCANNING_DISABLED` to `true` for a deployment named `sensor`:
+
[source,terminal,subs="+quotes"]
----
$ kubectl set env deploy/sensor ROX_DELEGATED_SCANNING_DISABLED="true"
----

Sensor should restart automatically. However, if it doesn't restart on its own, you should manually cause it to restart for the changes to take effect.

> If you're using the {product-title-short} Operator, there is a possibility that this change can be undone. In such a case, you should follow the specific instructions of the Operator.

Author: Removed entire section about environment variables.

=== Configuring delegated image scanning

A new delegated registration configuration specifies the registries from which image queries are to be delegated. For images observed by Sensor, this configuration allows you to delegate scans from no registries (default), all registries, or specific registries. To enable delegation of scans using the `roxctl` CLI, the Jenkins plugin, or the API, you must also specify a destination cluster and source registry.

.Prerequisites

* The environment variables must be configured. For more information on how to configure environment variables, see xref:../operating/examine-images-for-vulnerabilities.adoc#configuring-environment-variables_examine-images-for-vulnerabilities[Configuring environment variables].

.Procedure

. In the {product-title-short} portal, navigate to *Platform Configuration -> Clusters*.
. In the *Clusters* view header, click *Manage delegated scanning*.
. In the *Delegated Image Scanning* page, provide the necessary information:
** *Delegate scanning for*: Choose the scope of image delegation. Select one of the following options:
*** None (default)
*** All registries
*** Specified registries

> ** *Select default cluster to delegate to*: From the drop-down list, select the name of the destination cluster to which the delegated scans should be sent.

Reviewer: @dcaravel 
The way this is worded, it sounds like this is required, but you said it's not.

Should we call out that this is optional, but explicitly state the effect of leaving it blank?

Reviewer 2: Yes, this is optional and we should call that out. There has been alot of confusion regarding the cluster IDs, it will be great to explain that here.  Anywhere a cluster id is mentioned we should qualify that its only for CLI/API requests only and is NOT used by sensor.

The [API docs](https://github.com/stackrox/stackrox/blob/1f0c53ab12388e8997ba5b574696b9c9aadfe665/proto/api/v1/delegated_registry_config_service.proto#L22-L58) explain in more detail what each setting does.

May even be worth separating how this feature behaves from a sensor perspective vs. a roxctl/api perspective, for each setting.

> ** Click *Add registry*. Specify the source registry and destination cluster details. You can add more than one registry if required.

Reviewer: @dcaravel 

Likewise, the way this is worded, it sounds like adding at least one row to this list is required, but you said it's not.

We should specify when you would want to add one or more lines here, and why you might leave the cluster portion of this set to none.

Reviewer 2: Agreed, see my response above. Believe adding more details on these settings and how sensor reacts separatly from how roxctl/API requests are handled is needed.

. Click *Save*.

> Image integrations are now synchronized between Central and Sensor, with Sensor capturing pull credentials for each namespace. Sensor then uses these credentials to authenticate image registrations.

Reviewer 2: ```suggestion
Image integrations are now synchronized between Central and Sensor, and Sensor captures pull secrets from each namespace. Sensor then uses these credentials to authenticate image registries.
```

=== Installing and configuring scanning on secured clusters

[id="using-the-operator_{context}"]
==== Using the Operator

{product-title-short} Operator installs a Scanner-slim version on each secured cluster to scan images in the {ocp} integrated registry and optionally other registries.

For more information, see xref:../installing/installing_ocp/install-secured-cluster-ocp.adoc#installing-sc-operator[Installing {product-title-short} on secured clusters by using the Operator].

[id="using-helm_{context}"]
==== Using Helm

Secured Cluster Services Helm chart (`secured-cluster-services`) installs a Scanner-slim version on each secured cluster.
In Kubernetes, the secured cluster services include Scanner-slim as an optional component.
On {ocp}, however, {product-title-short} installs a Scanner-slim version on each secured cluster to scan images in the {ocp} integrated registry and optionally other registries.

* For {ocp} installations, see xref:../installing/installing_ocp/install-secured-cluster-ocp.adoc#installing-secured-cluster-services-quickly_install-secured-cluster-ocp[Installing the secured-cluster-services Helm chart without customization].

> * For non-{ocp} installations footnote:disclaimer[Platforms such as Amazon Elastic Kubernetes Service (Amazon EKS), Google Kubernetes Engine (Google GKE), and Microsoft Azure Kubernetes Service (Microsoft AKS).], see xref:../installing/installing_other/install-secured-cluster-other.adoc#installing-secured-cluster-services-quickly_install-secured-cluster-other[Installing the secured-cluster-services Helm chart without customization].

Reviewer 3: [Per the ISG](https://www.ibm.com/docs/en/ibm-style?topic=references-footnotes) we typically should avoid footnotes. Is it possible to include these examples either in the sentence or as a NOTE under this bullet point?

[id="verifying-after-installation_{context}"]
==== Verifying after installation

.Procedure

* Verify that the status of the secured cluster indicates that Scanner is present and healthy:
** In the {product-title-short} portal, navigate to *Platform Configuration -> Clusters*.
** In the *Clusters* view, select a cluster to view its details.

> ** In the *Health Status* card, ensure that *Scanner* is present and is marked as *Healthy.*

Reviewer 3: Although it's a good call to make the main step unordered because it's a single step, these nested sub-steps [should be ordered because the sequence of actions is important](https://www.ibm.com/docs/en/ibm-style?topic=format-lists#ordered-lists).

[id="additional-resources_examine-images-for-vulnerabilities"]
[role="_additional-resources"]
== Additional resources

> * For more information about Common Vulnerabilities and Exposures (CVEs), see the link:https://access.redhat.com/security/cve[Red Hat CVE Database].

Reviewer 4: Per OpenShift documentation guidelines for [Additional resources sections](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#additional-resources-sections).

> You must not include paragraphs in the section. Use an unordered list.

```suggestion
* link:https://access.redhat.com/security/cve[Red Hat CVE Database].
```

Author: Thanks for your feedback, but I am afraid I have not worked on this piece of content. The reason you see it under changed files is perhaps because of the removal of white spaces.

Reviewer 4: Thanks, Agil! 
Maybe a consideration for another time if you update these files :)
